presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeExecution.java

```diff
@@ -87,4 +93,30 @@ public void testNativeExecutionShuffleManager()
         assertEquals(0, shuffleId);
         assertEquals(shuffleManager.getNumOfPartitions(shuffleId), bypassMergeSortShuffleHandle.numMaps());
     }
+
+    // TODO: re-enable the test when presto native extends support beyond builtin functions.
+    @Ignore
+    /**
+     * Test native execution of a cpp function declared via json file.
+     */
+    public void testNativeExecutionWithJsonFileBasedFunction()
+    {
+        Session session = Session.builder(getNativeSession()).build();
+        PrestoSparkQueryRunner queryRunner = (PrestoSparkQueryRunner) getQueryRunner();
+        setupJsonFunctionNamespaceManager(queryRunner);
```

Comment:
Can we move this into shared setup logic?

Reply:
At this point, it needs a resource via property `json-based-function-manager.path-to-function-definition` and I'm checking in `src/test/resources/eq.json` to presto-native-execution module. If the recommendation is to pull setup logic to AbstractTestQueryFramework, the setup will apply to several presto maven modules.

If the setup is specific to the native execution test here, what's the advantage of moving this into shared setup logic?

Reply:
Setup logic common to Presto-on-Spark-on-Velox is in com.facebook.presto.spark.AbstractTestPrestoSparkQueries. I think we can move this configuration there, no?